src/Analyzers/Core/CodeFixes/PopulateSwitch/AbstractPopulateSwitchCodeFixProvider.cs

```diff
@@ -58,8 +58,8 @@ public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var diagnostic = context.Diagnostics.First();
             var properties = diagnostic.Properties;
-            var missingCases = bool.Parse(properties[PopulateSwitchStatementHelpers.MissingCases]);
-            var missingDefaultCase = bool.Parse(properties[PopulateSwitchStatementHelpers.MissingDefaultCase]);
+            var missingCases = bool.Parse(properties[PopulateSwitchStatementHelpers.MissingCases]!);
```

Comment:
>! [](start = 97, length = 1)

We typically reserve the `!` suppressions for things that are checked in the current method (directly or via straightforward helper method) but the compiler is unable to track.
For more complicated invariants, we use assertions instead (`Debug.Assert(... is object);` or `is { }` depending on preference).
These two (and two more below in this file) strike me as cases for assertions.

[https://github.com/dotnet/roslyn/blob/master/docs/contributing/Nullable%20Annotations.md](https://github.com/dotnet/roslyn/blob/master/docs/contributing/Nullable%20Annotations.md)

Reply:
An assert seems unnecessary here since `bool.Parse()` will throw an exception if the argument is `null`.

---
In reply to: [405655126](https://github.com/dotnet/roslyn/pull/43023#discussion_r405655126) [](ancestors = 405655126)

Reply:
Ya. The ! was just to keep the compiler warning quiet so that the original behavior of throwing from bool.Parse would be preserved.

Reply:
Makes sense. Thanks